src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditCategory.java

```diff
@@ -0,0 +1,44 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.impl;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+
+public enum AuditCategory {
+    BAD_HEADERS,
+    FAILED_LOGIN,
+    MISSING_PRIVILEGES,
+    GRANTED_PRIVILEGES,
+    OPENDISTRO_SECURITY_INDEX_ATTEMPT,
+    SSL_EXCEPTION,
+    AUTHENTICATED,
+    COMPLIANCE_DOC_READ,
+    COMPLIANCE_DOC_WRITE,
+    COMPLIANCE_EXTERNAL_CONFIG,
+    COMPLIANCE_INTERNAL_CONFIG_READ,
+    COMPLIANCE_INTERNAL_CONFIG_WRITE;
+
+    private static Logger log = LogManager.getLogger(AuditCategory.class);
+
+    public static EnumSet<AuditCategory> parse(final List<String> categories) {
+        EnumSet<AuditCategory> set = EnumSet.noneOf(AuditCategory.class);
+        if (categories == null)
+            return set;
+
+        categories
+                .stream()
+                .filter(Objects::nonNull)
+                .map(category -> category.toUpperCase())
+                .forEach(category -> {
```

Comment:
nit: use `stream` API:
```
        return categories
                .stream()
                .filter(Objects::nonNull)
                .map(String::toUpperCase)
                .map(category -> {
                    try {
                        return AuditCategory.valueOf(category);
                    } catch (IllegalArgumentException ex) {
                        log.error("Invalid category type: {} Exception: {}", category, ex.getMessage());
                        return null;
                    }
                })
                .filter(Objects::nonNull)
                .collect(Collectors.toCollection(() -> set));
```

Reply:
I ve considered using this but i felt it was not efficient to loop through the list twice again to filter and convert to set. I would prefer more traditional approach for doing this (removing stream api itself). What do you think?

Reply:
There will be no additional iterations over the stream (the only one is when it is consumed in the `collect()`), so it would be as efficient as `forEach()`.

Reply:
Yes. I thought of ignoring the second loop. 
I have updated to use stream api